### ISSUE TITLE
Documentation for aliases and redirection

### ISSUE THREAD

**Issue by scottsweb** [humanmade/Mercator#34]
I have been testing Mercator on a new multisite setup and would like to see some more detailed documentation around aliases and the correct way to set a primary domain. At times I see sites enter into a redirect loop and think I am most likely doing_it_wrong.

Right now there are 4 places a domain name can be set within the network admin:

* The info tab (which I am guessing has to remain as the sub-domain/sub-directory on a multisite e.g. mysite.multisite.com)
* The site settings tab (which allows you to modify two site options for `siteurl` and `home`. I am guessing these should be set to the primary domain, e.g. mysite.com).
* The aliases tab (which allows you to map other domains to your site. e.g. mysite.com and mysiterocks.com. If mysite.com is set as primary in `siteurl` and `home` I would have thought that mysiterocks.com would redirect but it does not. All aliases work standalone).

So my questions are:

1. What is the correct way to set a new primary domain
2. How do I get other domains to redirect to the primary domain to avoid duplicate content

Thanks

**Comment by robrecord** [comment]
I would like to know too, please.

**Comment by robrecord** [comment]
Without documentation I don't know what settings I should be using to set up my sites; i had read in earlier comments that I should change the site url in Sites Info tab to the primary domain I want to use for that site. Anything else that should be allowed to access the site should be in aliases. Is this correct?

Also I don't know how to get Mercator to redirect aliases to the primary domain. The database info that switched between "primary" and not-primary in wp-domain-mapping is now being used for "Active" or non-active - not really sure what Active means in this case. Could you add some context-based help or a small description?

Many thanks.

**Comment by mrteera** [comment]
:+1:

**Comment by guillaumemolter** [comment]
I would like to know too...I heard goof things about Mercator...but so far I haven't been able to make it "work properly" and probably because I'm doing it wrong too.

**Comment by mrjarbenne** [comment]
Did you note the additional instructions added to the readme file recently?

**Comment by guillaumemolter** [comment]
Yeah but this is not answering to the questions from @robrecord and @scottsweb . In my case unless I'm doing something wrong Mercator doesn't hide the original URL, websites are therefore accessible via 2 URLs...which mean duplicate content and terrible SEO (unless again I'm wrong and this should be documented).

**Comment by mrjarbenne** [comment]
Yes. That's correct. You aren't doing anything wrong. Currently Mercator just adds an additional alias for the domain. It doesn't redirect to the mapped domain from the unmapped domain. Obviously that's the big missing piece here: one that has been raised already in issue #3

**Comment by guillaumemolter** [comment]
Indeed :-). This plugin was "sold" to me as a great replacement for the original domain mapping plugin. However while it looks promissing I think I will need to keep keep *using outdated practices and the non-sence* while mercator becomes a real alternative.

**Comment by mrjarbenne** [comment]
I moved my site back as well.

**Comment by mrjarbenne** [comment]
It would be great if @rmccue could chime in. I note that the are using this on Happytables.com according to a recent update to the readme. There must be a way to do this if humanmade is using this at scale. Perhaps additional redirect rules beyond the scope of the plugin that makes it all work.

**Comment by guillaumemolter** [comment]
That would be awesome...

**Comment by robrecord** [comment]
I believe I have this type of redirect working with my nginx server. I think it came "out of the box" with my installation of Easy Engine, which sets up nginx and Wordpress among others. when I get some more time I'll try to prise out the rules that allow this behaviour to work.

**Comment by rmccue** [comment]
Hi, sorry for not chiming in earlier. :)

Happytables is pretty custom, and the way we mark a domain as primary right now is terribly hacky. Essentially, HT only allows one-alias-per-site, despite Mercator allowing more under the hood. We then determine if they have an alias, and redirect users to that.

This isn't a particularly scalable solution, and doesn't fit with the way I want Mercator to work.

With that said, I have been working on primary domain support, so we may be able to launch that soon.

**Comment by mrjarbenne** [comment]
Thanks for the update. And for all your work on this.

**Comment by guillaumemolter** [comment]
Thank you! Look forward for the update :-)

**Comment by scarstens** [comment]
@scottsweb @robrecord 
to circle back on proper setup, from what I've read in the code (which I've now completely reviewed in whole) is that the plugin was intended to be used `without changing the site_url in the database`. This is confusing, especially when trying to super-admin all the sites and not being able to see the actual urls. 

The redirect loop occurs when you have SSO enabled and change the site_url (see the other open issues regarding the redirect loop issue).

This might also be a dupe of this ticket: https://github.com/humanmade/Mercator/issues/3

**Comment by roborourke** [comment]
@scarstens @scottsweb @mrjarbenne @robrecord @guillaumemolter so some updates here:

Redirection has been a bit of a pain point to implement properly so there is now a separate plugin as part of the imminent v2 updates:

https://github.com/humanmade/Mercator-Redirect

Follow the guide there for adding support for v1.0.0 of Mercator using the `legacy` redirection eg. the first alias is the primary domain.

In future Mercator will treat the `site_url` as the primary domain and the new GUI plugin will handle updating that:

https://github.com/humanmade/Mercator-GUI

Currently there is a new method on the `Mapping` object called `make_primary()` that will update the `site_url` and also add the current value there as a replacement alias.

The Redirection plugin will handle some redirection options you might want such as forcing admin to use the main network URL to take advantage of a wildcard SSL cert for example.
